### DISCUSSION TITLE
Set current logged in user as default value for User lookup field in the form

### DISCUSSION THREAD

**Discussion by ziaatrix** [serenity-is/Serenity#7129] [General]
Hi,

How to set current logged in user as default value for User lookup field in the form?

**Comment by VictorTomaili** [comment]
on load form, check the lookup items which matches the userdefinition value (id field, email, username etc. which one can be applied and unique depends on your project) and set the id value of user to the lookupeditor value. ScriptUserDefinition and UserDefinition are different. Please consider this. 

```ts
import { userDefinition } from "./Authentication/Authorization";
```

**Reply by ziaatrix** [reply]
Hi Victor,

Thank you for your reply. This library is not available in our project.
![image](https://github.com/serenity-is/Serenity/assets/118987818/d2051482-139e-4fd5-b3e5-e2e183624312)
The only available is the Authorization which is ScriptUserDefinition
![image](https://github.com/serenity-is/Serenity/assets/118987818/2130e24e-2114-45b9-b8ae-0ac4513c18b6)

**Reply by VictorTomaili** [reply]
It is same thing I guess, just the difference between the serenity versions.

**Reply by ziaatrix** [reply]
Hi Victor,

I used the same (ScriptUserDefinition) however, it is not retrieving id. It only retrieves this much user data. We need to get the user id so that we can set it as default value for my userlookup field.

![image](https://github.com/serenity-is/Serenity/assets/118987818/7b6ca2a4-da56-452d-ac7a-115b3bd0b0c8)

 


Is there any other way like using the DefaultValue attribute in Row.cs?

**Reply by VictorTomaili** [reply]
add userId to your scriptuserdefinition 
https://github.com/serenity-is/serene/blob/f65a992d70f83101720b196b0dcd51b95caf38ac/src/Serene.Web/Modules/Administration/User/UserDataScript.cs#L37
